R/read_multi.R

```diff
@@ -0,0 +1,89 @@
+#' @rdname read_multi
+#'
+#' @title Read spectral data from multiple files
+#'
+#' @description
+#' Wrapper functions for reading files in batch. 
+#'
+#' @details
+#' \code{read_any()} provides a single function to quickly read
+#' in any of the supported formats, it assumes that the file 
+#' extension will tell it how to process the spectra.
+#' \code{read_zip()} provides functionality for reading in
+#' spectral map files with ENVI file format or as individual files
+#' in a zip folder. If individual files, spectra are concatenated.
+#'
+#' @param file file to be read from or written to.
+#' @param \ldots further arguments passed to the submethods.
+#'
+#' @return
+#' All \code{read_*()} functions return OpenSpecy objects
+#'
+#' @examples
+#' read_any(read_extdata("raman_hdpe.csv"))
+#' read_any(read_extdata("ftir_ldpe_soil.asp"))
+#' read_any(read_extdata("ftir_ps.0"))
+#' read_zip(read_extdata("testdata_zipped.zip"))
+#' read_zip(read_extdata("CA_tiny_map.zip"))
+#'
+#' @author
+#' Zacharias Steinmetz, Win Cowger
+#'
+#' @seealso
+#' \code{\link{read_spec}()}
+#' 
+#' @export
+read_any <- function(file, ...) {
+  if (grepl("(\\.csv$)|(\\.txt$)", ignore.case = T, file)) {
+    os <- read_text(file = file, ...)
+  } else if (grepl("\\.[0-999]$", ignore.case = T, file)) {
+    os <- read_opus(file = file, ...)
+  } else if (grepl("(\\.asp$)|(\\.spa$)|(\\.spc$)|(\\.jdx$)",
+                   ignore.case = T, file)) {
+    ex <- strsplit(basename(file), split="\\.")[[1]][-1]
+    os <- do.call(paste0("read_", tolower(ex)), list(file = file, ...))
+  } else if (grepl("(\\.zip$)", ignore.case = T, file)) {
+    os <- read_zip(file = file, ...)
+  } else {
+    os <- read_spec(file = file, ...)
+  }
+
+  return(os)
+}
+
+#' @rdname read_multi
+#'
+#' @importFrom utils unzip
+#' @importFrom hySpc.read.ENVI read_ENVI_Nicolet
```

Comment:
If we want to submit v1.0.0 to CRAN, we can't use the hySpc.read.ENVI package as long it's not on CRAN itself. We could either motivate them to submit soon or implement those functions on our own.

Reply:
> If we want to submit v1.0.0 to CRAN, we can't use the hySpc.read.ENVI package as long it's not on CRAN itself. We could either motivate them to submit soon or implement those functions on our own.

@wincowgerDEV was there any specific reason to prefer hySpc.read.ENVI over caTools (https://cran.r-project.org/web/packages/caTools/index.html). caTools is on CRAN and can also read ENVI files.

Reply:
Hyperspec has been overhauling their package structure and it ended up breaking the reading function we were previously using. Id be happy with using a different package, probably more stable at the moment.